File: RepoTransBench/target_projects/Python/C++/maxcountryman_flask-login/tests/test_config.cpp
#include <gtest/gtest.h>
#include "src/config.h"

using namespace flask_login_config;

TEST(ConfigTest, ConfigValues) {
    EXPECT_EQ(COOKIE_NAME, "remember_token");
    // days==365
    auto days = std::chrono::duration_cast<std::chrono::hours>(COOKIE_DURATION).count() / 24;
    EXPECT_EQ(days, 365);
    EXPECT_EQ(COOKIE_SECURE, false);
    EXPECT_EQ(COOKIE_HTTPONLY, true);
    EXPECT_TRUE(EXEMPT_METHODS.count("OPTIONS") == 1);
    EXPECT_EQ(LOGIN_MESSAGE, "Please log in to access this page.");
    EXPECT_EQ(LOGIN_MESSAGE_CATEGORY, "message");
    EXPECT_EQ(REFRESH_MESSAGE, "Please reauthenticate to access this page.");
    EXPECT_EQ(REFRESH_MESSAGE_CATEGORY, "message");
    EXPECT_EQ(ID_ATTRIBUTE, "get_id");
    EXPECT_TRUE(SESSION_KEYS.count("_user_id"));
    EXPECT_TRUE(SESSION_KEYS.count("_remember"));
    EXPECT_EQ(USE_SESSION_FOR_NEXT, false);
}

File: RepoTransBench/target_projects/Python/Java/maxcountryman_flask-login/src/test/java/com/maxcountryman/flasklogin/original/TestConfig.java
package com.maxcountryman.flasklogin.original;

import com.maxcountryman.flasklogin.config.Config;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestConfig {

    @Test
    public void testConfigValues() {
        assertEquals("remember_token", Config.COOKIE_NAME);
        assertEquals(365, Config.COOKIE_DURATION.toDays());
        assertFalse(Config.COOKIE_SECURE);
        assertTrue(Config.COOKIE_HTTPONLY);
        assertTrue(Config.EXEMPT_METHODS.contains("OPTIONS"));
        assertEquals("Please log in to access this page.", Config.LOGIN_MESSAGE);
        assertEquals("message", Config.LOGIN_MESSAGE_CATEGORY);
        assertEquals("Please reauthenticate to access this page.", Config.REFRESH_MESSAGE);
        assertEquals("message", Config.REFRESH_MESSAGE_CATEGORY);
        assertEquals("get_id", Config.ID_ATTRIBUTE);
        assertTrue(Config.SESSION_KEYS.contains("_user_id"));
        assertTrue(Config.SESSION_KEYS.contains("_remember"));
        assertFalse(Config.USE_SESSION_FOR_NEXT);
    }
}